### ISSUE TITLE
bug: combination of selector-pseudo-element-colon-notation selector-pseudo-element-no-unknown

### ISSUE THREAD

**Issue by goldylucks** [stylelint/stylelint#2208]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?
Bug in `.stylelintrc`

> Which rule, if any, is this issue related to?
selector-pseudo-element-no-unknown

> What CSS is needed to reproduce this issue?
```css
:global .foo {
  color: pink;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "selector-pseudo-element-no-unknown": [true, {
      ignorePseudoElements: ["/global/"]
    }]
  }
}
```

> Which version of stylelint are you using?
`7.7.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?
CLI with `stylelint 'web/**/*.css'

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?
no

> What did you expect to happen?
no errors to be flagged

> What actually happened (e.g. what warnings or errors you are getting)?
```shell
Unexpected unknown pseudo-class selector ":global"   selector-pseudo-class-no-unknown
```
The following warnings were flagged:
```shell
web/containers/Auth/Auth.css
 22:1  ✖  Unexpected unknown pseudo-class selector ":global"   selector-pseudo-class-no-unknown
```

## PR with failing test
#2209

**Comment by m-allanson** [comment]
@goldylucks thanks for using the issue template. 

I'm unable to replicate this issue. Running `./bin/stylelint.js "failing.css"` on your PR outputs the following:

```shell
failing.css
 1:1  ✖  Expected double colon pseudo-element notation   selector-pseudo-element-colon-notation
```

which seems correct based on the config you've included there.

Are you able to create a standalone repo that demonstrates the problem?

**Comment by goldylucks** [comment]
@m-allanson sry about that
forgot some local changes
pushed new PR

let me know if it's clearer now

**Comment by m-allanson** [comment]
@goldylucks thanks for the updated PR. The error you're seeing is from the rule [`selector-pseudo-class-no-unknown`](http://stylelint.io/user-guide/rules/selector-pseudo-class-no-unknown/) (note _class_, not _element_) which is enabled in `stylelint-config-standard`.

You can remove the error by updating your config with:
```
"selector-pseudo-class-no-unknown": [true, {
  ignorePseudoClasses: ["global"]
}]
```

**Comment by goldylucks** [comment]
@m-allanson thanks for pointing that out
I already tried that tho, still same error occurs :(

any other ideas?

**Comment by evilebottnawi** [comment]
@goldylucks provide your version `node`, `npm`, `stylelint`, `package.json`, `stylelint configuration`, maybe i can help you.

**Comment by goldylucks** [comment]
Thanks @evilebottnawi !
here r details requested. Lemme know if there's anything I missed

```bash
$ npm -v
3.10.8

node -v
v6.9.1

$ cat package.json | grep \"stylelint\"
"stylelint": "^7.7.0",

$ cat .stylelintrc 
{
  "extends": "stylelint-config-standard",
  "rules": {
    "declaration-block-properties-order": "alphabetical",
  }
}

```

**Comment by evilebottnawi** [comment]
@goldylucks 
Try:
> .stylelintrc
```json
{
    "extends": "stylelint-config-standard",
    "rules": {
        "declaration-block-properties-order": "alphabetical",
        "selector-pseudo-class-no-unknown": [true, {
            "ignorePseudoClasses": ["global"]
        }]
    }
}
```

**Comment by goldylucks** [comment]
@evilebottnawi ,
I tried that, following this [comment](https://github.com/stylelint/stylelint/issues/2208#issuecomment-269245751) by @m-allanson

**Comment by m-allanson** [comment]
@goldylucks your code samples are all slightly different but, as far as I can tell, you have the following. 

A config like:

```json
{
  "rules": {
    "selector-pseudo-class-no-unknown": true,
    "selector-pseudo-element-no-unknown": [
      true,  {
        "ignorePseudoElements": ["global"]
      }]
  }
}
```

CSS like: 
```css
:global .foo {
  color: pink;
}
```

And a warning from stylelint like:

```
Unexpected unknown pseudo-class selector ":global" (selector-pseudo-class-no-unknown)
```

If that's the case, then I think stylelint is working as expected. The warning is from the first rule `selector-pseudo-class-no-unknown` which is unaffected by the second rule `selector-pseudo-element-no-unknown`.

If that's not the issue you're seeing, then I think the best thing to do is to create a completely bare-bones repo that demonstrates the issue. The only dependency should be `stylelint`, and the config should only contain rules that trigger the problem. 

Strip out everything that isn't related to your issue. If there's rules in `stylelint-config-standard` that trigger the issue, copy them from [that config](https://github.com/stylelint/stylelint-config-standard/blob/master/index.js) into your demo config. But only the rules that are required for demonstrating the problem.

If you can create a setup that's laser-focused on the issue, we can hopefully get to the bottom of this.

You can use [this repo as an example](https://github.com/m-allanson/stylelint-issue-1693).

**Comment by goldylucks** [comment]
thanks @m-allanson 

the following configuration solved my problem:

```json
{
  "extends": "stylelint-config-standard",
  "rules": {
    "declaration-block-properties-order": "alphabetical",
    "selector-pseudo-class-no-unknown": [true, {
      "ignorePseudoClasses": ["global"]
    }],
    "selector-pseudo-element-no-unknown": [true, {
      "ignorePseudoElements": ["global"]
    }]
  }
}
```
